### PR TITLE
strict sphinx modelsrepository

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/33540

### CODE REVIEW

**Comment by kristapratico** [comment]
Failing mypy/pylint tracked in https://github.com/Azure/azure-sdk-for-python/issues/31601 https://github.com/Azure/azure-sdk-for-python/issues/32822

File: sdk/modelsrepository/azure-iot-modelsrepository/azure/iot/modelsrepository/_client.py [hunk 1]
```diff
@@ -45,25 +45,25 @@ class ModelsRepositoryClient(object):
 
     def __init__(self, **kwargs):  # pylint: disable=missing-client-constructor-parameter-credential
         # type: (Any) -> None
-        """
+        """For additional request configuration options, please see [core options](https://aka.ms/azsdk/python/options).
```
Comment (annatisch) [inline sdk/modelsrepository/azure-iot-modelsrepository/azure/iot/modelsrepository/_client.py:48]: I feel like the first line of the docstring should be something more like: Client for the Models Repository service (or something like that - maybe there's a line from the README we can use.....)

Comment (kristapratico) [reply sdk/modelsrepository/azure-iot-modelsrepository/azure/iot/modelsrepository/_client.py:48]: good point, let me fix that

**Review by annatisch — APPROVED** [review]

**Pull request merged**